test.py

```diff
@@ -0,0 +1,139 @@
+import logging
+import signal
+import time
+from threading import Lock, Thread
+from typing import Callable, Literal
+import pygame
+from luma.emulator.device import pygame as luma_pygame
+from PIL import ImageFont
+from display import ScreenDisplay, Terminal
+from sensor_main import Device
+from util import CONFIG, ConfigManager, FileLock
+
+def main():
+    logging.basicConfig(
+        format='%(asctime)s %(levelname)-8s %(message)s',
+        level=logging.DEBUG,
+        datefmt='%Y-%m-%d %H:%M:%S',
+        filename='sensor.log',
+        encoding='utf-8',
+    )
+    CONFIG["config_file"] = "sensor_specs/sensors_config.ini"
+    CONFIG["config_lock"] = "sensor_specs/envs.lock"
+    #pylint: disable=protected-access
+    ConfigManager._config_cache["config_file"] = "sensor_specs/sensors_config.ini"
+    ConfigManager._config_cache["file_lock"] = FileLock("sensor_specs/envs.lock")
+    #pylint: enable=protected-access
+    pygame.init()
+    emulator = PygameEmulator(320, 240, rotate=0)
+    keyboard = PigpioWrapper()
+    device = Device(
+        pi_gpio=keyboard,
+        display=ScreenDisplay(
+            terminal=Terminal(
+                device=emulator,
+                font=ImageFont.truetype("DejaVuSansMono.ttf", 21)
+            )
+        )
+    )
+    stop = False
+
+    def sigint_handler(_1, _2):
+        nonlocal stop
+        stop = True
+        device.stop()
+
+    signal.signal(signal.SIGINT, sigint_handler)
+    device_thread = Thread(target=device.run)
+    device_thread.start()
+
+    while not stop:
+        time.sleep(0.02)
+        pygame.event.pump()
+        keystate = pygame.key.get_pressed()
+        if keystate[pygame.K_ESCAPE] or pygame.event.peek(pygame.QUIT):
+            break
+        for callback in keyboard.callbacks:
+            if keyboard.read(callback.pin) != callback.last_state:
+                callback.change_state()
+
+    device.stop()
+    device_thread.join()
+    emulator.abort()
+    pygame.quit()
+
+PIN_TO_KEY = {
```

Comment:
Can we move const declaration to the top? If not, please add whitespace before.
Please consider.